manifests/charts/istio-control/istio-discovery/files/waypoint.yaml

```diff
@@ -173,13 +173,12 @@ spec:
           successThreshold: 1
           timeoutSeconds: 1
         securityContext:
-          privileged: true
+          privileged: false
           runAsGroup: 1337
           runAsUser: 0
```

Comment:
```suggestion
          runAsUser: 1337
```

Reply:
With `runAsUser` it becomes a bit tricky since we need to change the mountpoints too.

For example, this is what I had done since we mount to  /var/run/
```
volumeMounts:
        - mountPath: /var/run/secrets/istio
          name: istiod-ca-cert
        - mountPath: /var/lib/istio/data
          name: istio-data
        - mountPath: /etc/istio/proxy
          name: istio-envoy
        - mountPath: /var/run/secrets/tokens
          name: istio-token
        - mountPath: /etc/istio/pod
          name: istio-podinfo
```

I have one question, How can one change the Istio CA ADDR path I tried looking at the environment variables for the pilot-agent but it wasn't clear

```
2023-09-01T10:09:36.467957Z    error    failed to start SDS server: failed to start workload secret manager failed to find root CA cert for CA: root CA file for
 CA does not exist var/run/secrets/istio/root-cert.pem
Error: failed to start SDS server: failed to start workload secret manager failed to find root CA cert for CA: root CA file for CA does not exist var/run/secret
s/istio/root-cert.pem
```
Thank you!

Reply:
There is a CA_ADDR env

Reply:
Apologies, I mean't how to set the directory on where Istio would store the root-cert.pem
```
2023-09-01T10:09:36.467957Z    error    failed to start SDS server: failed to start workload secret manager failed to find root CA cert for CA: root CA file for
 CA does not exist var/run/secrets/istio/root-cert.pem
```

Reply:
My colleague helped me it's `OUTPUT_CERTS` now I need to fix the spiffie paths

Reply:
Is the workload-spiffe socket hardcoded?
```
2023-09-08T06:55:34.742027Z    info    cache    Root cert has changed, start rotating root cert
2023-09-08T06:55:34.742043Z    info    ads    XDS: Incremental Pushing:0 ConnectedEndpoints:0 Version:
2023-09-08T06:55:34.742359Z    info    cache    returned workload trust anchor from cache    ttl=23h59m59.257644141s
2023-09-08T06:55:35.543046Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
2023-09-08T06:55:35.543132Z    error    sds    SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/w
2023-09-08T06:55:37.544165Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
2023-09-08T06:55:37.544253Z    error    sds    SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/w
2023-09-08T06:55:41.545291Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
2023-09-08T06:55:41.545357Z    error    sds    SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/w
2023-09-08T06:55:49.546404Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
2023-09-08T06:55:49.546484Z    error    sds    SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/w
2023-09-08T06:56:04.899382Z    info    Status server has successfully terminated
```
Found this here
```
pkg/security/security.go
48:     // WorkloadIdentitySocketPath is the well-known path to the Unix Domain Socket for SDS.
49:     WorkloadIdentitySocketPath = "./var/run/secrets/workload-spiffe-uds/socket"
```

Reply:
Inside the waypoint pod it's not `bookinfo-productpage-istio-waypoint-54649856d-wx6ln`
```
2023-09-12T17:25:37.999261Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
 permission denied
 ```

But in the gateway pod it is `istio-system/bookinfo-gateway-istio-f474476-26wxk`
```
2023-09-12T17:30:26.730066Z    warn    Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds:
 read-only file system
```